File: LeetCode_02/Solution_ZhongJing.kt
class ListNode(var `val`: Int) {
    var next: ListNode? = null
}

/**
 * 日期：2023-07-02
 * 作者：仲景
 */
class Solution {
    fun addTwoNumbers(l1: ListNode?, l2: ListNode?): ListNode? {
        val newHead = ListNode(-1)
        var node = newHead
        var node1 = l1
        var node2 = l2
        var num = 0

        while (node1 != null || node2 != null) {
            val sum = (node1?.`val` ?: 0) + (node2?.`val` ?: 0) + num
            num = if (sum >= 10) 1 else 0
            node.next = ListNode(sum % 10)
            node = node.next!!
            node1 = node1?.next
            node2 = node2?.next
        }

        if (num > 0) {
            node.next = ListNode(1)
        }

        return newHead.next
    }
}

File: LeetCode_09/LeetCode_09_04/Solution_ZhongJing.py
"""
日期：2023年08月04日
作者：仲景
"""


class Solution:
    def isPalindrome(self, x: int) -> bool:
        # 如果是负数，一定不是回文数
        if x < 0:
            return False
        
        num = x
        reverseNum = 0
        while num > 0:
            reverseNum = reverseNum * 10 + num % 10
            num //= 10
        
        return reverseNum == x